Moved to a more flexible regex match #9

balneaves opened the pull request:
Switch to using regex so it can support the suggest patterns

Case insensitive:
^\W_(TODO\s_:|TODO|FIXME\s_:|FIXME)\s_(.*)$

Resolves #9 
Resolves #14

balneaves committed:
Moved to a more flexible regex match #9

maximeplante commented:
Your regex does not seem to be really flexible. I tested it against a few cases and only one of them worked.
You can take a look here : https://regex101.com/r/cL7nX8/4
I used cases with and without slashes since I don't know if the regex is supposed to match the comment markers or not.

---

I modified you regex a bit to make it more general.
`^\/\/\s*(TODO\s*:|TODO\s*|FIXME\s*:|FIXME)\s*(.*)$` with the `i` modifier to make it case-insensitive.
You can test it here : https://regex101.com/r/kC4yK7/1

Hope this helps!

Cryszon commented:
It still doesn't capture other comment delimiters, such as `/* */` or `#`. I guess it's a question of how liberal you'd want the matching to be. To capture "everything" something as simple as
`/^.*(TODO|FIXME)\W*(.*)$/gim` would suffice.

balneaves commented:
@Max840 - Sorry the regex in the code is slightly different from what I put in the pull request comment, should have read `^\W*(TODO\s*:|TODO|FIXME\s*:|FIXME)\s*(.*)$` (with the `i` modified)

@Cryszon Yeah, it ignores the comment part entirely for the sake of performance/simplicity... just expects TODO/FIXME to be the first word-characters on that line.  So in theory it would match a json line of `"todo": "test"` for example

Cryszon commented:
As a crude workaround, you could limit the amount of non-word, non-whitespace characters between the identifier and the message to 1. This would allow matching of something like `TODO - Do stuff` or `TODO: Something` while ignoring `"todo": "test"` or `$todo->test()`, since both have more than 1 special character between. In addition `.` character could also be ignored to make sure `todo.test()` doesn't get matched. It could even be taken further by strictly ignoring only a delimiting `.` without any whitespace to allow `TODO . Stuff` to match, although I don't think that's widely used format anyway.

balneaves commented:
Little better https://regex101.com/r/iO8hM3/3 ?

maximeplante commented:
Looks good to me

Cryszon commented:
Slightly updated to `^\W*(?:TODO|FIXME)[ \t]*[^\w\r\n\f]{0,1}[ \t]*([\w\d].*)$` to ignore matching newlines. Updated version with a newline test case [here](https://regex101.com/r/iO8hM3/5).

balneaves commented:
@Cryszon Thanks, might keep the previous one though for simplicity, the extension seems to loop through line-by-line so shouldn't need to deal with newlines.  Guessing it does that so that it knows the line number, but I haven't looked too deep into the rest of this extension!

balneaves committed:
Updated regex expression to be slightly more accurate
Fixed extension for January release

balneaves commented:
I updated the regex, and fixed it for the next vscode insider release (which seems to have blocked access to the private API members that were being used)

balneaves commented:
@MattiasPernhult Any thoughts on this?  This extension doesn't work as of the latest update.

mikemimik commented:
I can confirm this extension is broken. I'm diving into the code to see if I can fix / optimize in anyway.

balneaves commented:
@mikemimik This pull request also fixes the extension for the latest vscode release I think... or at least it did last time I checked

mikemimik commented:
@salerth I'm in the process of a complete rewrite, incorporating the regex and changes from this PR.

MattiasPernhult commented:
This pull request is now merged into the master branch. Thanks a lot for that! 

And I'm sorry for my absence here, but I have had a lot to do in my final year of my bachelor degree.
